profitbricks-rest/src/test/java/org/apache/jclouds/profitbricks/rest/features/FirewallApiLiveTest.java

```diff
@@ -0,0 +1,185 @@
+import org.apache.jclouds.profitbricks.rest.ids.ServerRef;
+import org.apache.jclouds.profitbricks.rest.internal.BaseProfitBricksLiveTest;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotNull;
+
+@Test(groups = "live", testName = "FirewallApiLiveTest")
+public class FirewallApiLiveTest extends BaseProfitBricksLiveTest {
+   
+   DataCenter dataCenter;
+   Server testServer;
+   Nic testNic;
+   FirewallRule testFirewallRule;
+  
+   @BeforeClass
+   public void setupTest() {
+      dataCenter = createDataCenter();
+      assertDataCenterAvailable(dataCenter);
+            
+      testServer = api.serverApi().createServer(
+         Server.Request.creatingBuilder()
+            .dataCenterId(dataCenter.id())
+            .name("jclouds-node")
+            .cores(1)
+            .ram(1024)
+            .build());
+      
+      assertNodeAvailable(ServerRef.create(dataCenter.id(), testServer.id()));
+            
+      testNic = nicApi().create(
+              Nic.Request.creatingBuilder()
+              .dataCenterId(dataCenter.id())
+              .serverId(testServer.id())
+              .name("jclouds-nic")
+              .lan(1)
+              .build());
+
+      assertNicAvailable(testNic);
+   }
+   
+   @AfterClass(alwaysRun = true)
+   public void teardownTest() {
+      if (dataCenter != null)
+         deleteDataCenter(dataCenter.id());
+   }
+     
+   @Test
+   public void testCreateFirewallRule() {
+      assertNotNull(dataCenter);
+            
+      testFirewallRule = firewallApi().createFirewallRule(
+              FirewallRule.Request.creatingBuilder()
+              .dataCenterId(dataCenter.id())
+              .serverId(testServer.id())
+              .nicId(testNic.id())
+              .name("jclouds-firewall")
+              .protocol(FirewallRule.Protocol.TCP)
+              .portRangeStart(1)
+              .portRangeEnd(600)
+              .build());
+
+      assertNotNull(testFirewallRule);
+      assertEquals(testFirewallRule.properties().name(), "jclouds-firewall");
+      assertFirewallRuleAvailable(testFirewallRule);
+   }
+   
+
+   @Test(dependsOnMethods = "testCreateFirewallRule")
+   public void testGetFirewallRule() {
+      FirewallRule firewallRule = firewallApi().getFirewallRule(dataCenter.id(), testServer.id(), testNic.id(), testFirewallRule.id());
+
+      assertNotNull(firewallRule);
+      assertEquals(firewallRule.id(), testFirewallRule.id());
+   }
+
+   @Test(dependsOnMethods = "testCreateFirewallRule")
+   public void testListRules() {
+      List<FirewallRule> firewalls = firewallApi().getRuleList(dataCenter.id(), testServer.id(), testNic.id());
+
+      assertNotNull(firewalls);
+      assertFalse(firewalls.isEmpty());
+      assertEquals(firewalls.size(), 1);
```

Comment:
Assert that the list contains the created element, instead of verifying its size.